frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx

```diff
@@ -248,6 +258,15 @@ export function ObjectDetailFn({
   );
   const canEdit = !!(isWritebackEnabled && table);
 
+  let deleteRow;
+  if (canEdit) {
+    deleteRow = () =>
+      deleteRowFromObjectDetail({
+        id: zoomedRowID,
+        table,
+      });
+  }
+
```

Comment:
Can we maybe just wrap it into `useCallback` or `useMemo` to make it more declerative? For instance:

```jsx
const handleDeleteRow = useCallback(() => {
   if (canEdit) {
      deleteRowFromObjectDetail({ ... })
   }
}, [ ... ])

// OR

const onDeleteRow = useMemo(() => {
   if (!canEdit) {
      return
   }
   deleteRowFromObjectDetail({ ... })
}, [...])
```

Reply:
Done